packages/tailwind/src/hooks/use-tailwind-styles.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import type { TailwindConfig } from "../tailwind";
+import { separateMediaQueriesFromCSS } from "../utils/css/media-queries/separate-media-queries-from-css";
+import { rulesFor } from "../utils/css/rules-for";
+import { quickSafeRenderToString } from "../utils/quick-safe-render-to-string";
+import { getCssForMarkup } from "../utils/tailwindcss/get-css-for-markup";
+import { useRgbNonSpacedSyntax } from "../utils/compatibility/use-rgb-non-spaced-syntax";
+import { cssToJsxStyle } from "../utils/compatibility/css-to-jsx-style";
+import { unescapeClass } from "../utils/compatibility/unescape-class";
+import { sanitizeRuleSelector } from "../utils/compatibility/sanitize-rule-selector";
+import { makeAllRulePropertiesImportant } from "../utils/compatibility/make-all-rule-properties-important";
+
+/**
+ * Gets all the necessary information from the node and the Tailwind config to be able
+ * to apply all the Tailwind styles.
+ */
+export function useTailwindStyles(
```

Comment:
This hook is meant to convert a React `node` and Tailwind `config` into a metadata object that contains the necessary stuff for the Tailwind component to do it's thing.

It does a few things:
1. Does a safe rendering of the node
2. Converts all `rgb` with spaced syntax (i.e. `rgb(239 68 68 / 1)' -> `rgb(239, 68, 68)`)
   - See https://github.com/resend/react-email/blob/canary/packages/tailwind/readme.md#rgb-syntax-color-changes
3. Using Regex, separates all media queries from normal CSS classes
4. Loops through all the CSS selectors/rules and starts aggregating their `JSX` styles into their appropriate class names
  - This also requires that we remove all Tailwind escaping from the class names
5. Loops through all media queries [sanitizing their rule selectors](https://github.com/resend/react-email/blob/canary/packages/tailwind/readme.md#the-treatment-for-media-query-class-names) and adding `!important` to all their properties
6. Removes all line-breaks and duplicated unnecessary spaces from the media queries
7. Collects all the class names used in the media queries

After that, it then returns:
- The classes that should not be inlined into a `style` prop (for a use that I'll later mention) which are right now only the classes used in media queries
- The media queries with sanitized selectors, that is, with their selectors having characters that don't need to be escaped to be used in a `<style>`
- A map of the JSX style object for each class